File: codeforces/CR#638D2/Phoenix and Balance/main.py
# WA
import math

t = int(input())
while(t>0):
	n = int(input())
	count = 0
	i = 1
	s1 = 0
	s2 = 0
	while(i+1<=n-2):
		mn = math.pow(2,i)
		mx = math.pow(2,i+1)
		s1+=mn
		s2+=mx

		i+=2

	mn = math.pow(2,i)
	mx = math.pow(2,i+1)
	s1+=mx
	s2+=mn

	print(abs(int(s1-s2)))

	t-=1

File: codeforces/CR#638D2/Phoenix and Balance/main.cpp
// WA
#include <bits/stdc++.h>
using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int t;
    cin>>t;
    while(t--){
    	int n;
    	cin>>n;
    	int count = 0;
    	long long int diff = 0;
    	int i=1;
    	while(i+1<=n){

    		if(count == 0) diff+=(pow(2, i) - pow(2, i+1));
    		else diff+=(pow(2, i+1) - pow(2, i));

    		count+=1;
    		count%=2;
    		i+=2;
    	}
    	cout<<abs(diff)<<endl;
    }
    return 0;
}
